src/client/pydaos/pydaos_core.py

```diff
@@ -383,5 +377,19 @@ def __contains__(self, key):
         except KeyError:
             return False
 
+    def __eq__(self, other):
+        # Not efficient for now. Could use the bulk operation.
```

Comment:
Idea: we could also implement a C version and call that, right? There has been some mild, loose discussion about the possibility of making some of the data mover core functions available elsewhere. This is somewhat related, as we could have C-level functions for "equal", "copy", etc.

Reply:
Agreed, moving it to the C part would accelerate it. I will look into that.